mount/temp_unix.go

```diff
@@ -41,10 +41,16 @@ func SetTempMountLocation(root string) error {
 
 // CleanupTempMounts all temp mounts and remove the directories
 func CleanupTempMounts(flags int) (warnings []error, err error) {
-	mounts, err := mountinfo.GetMounts(mountinfo.PrefixFilter(tempMountLocation))
+	target, err := CanonicalizePath(tempMountLocation)
```

Comment:
This could just call the eval symlinks on `SetTempMountLocation` rather than for each call here.

Reply:
We can't eval symlinks to something that doesn't exist yet.

Reply:
Updated PR. _Ideally_, we should do this everywhere (perform canonicalization during in the beginning, not just before unmount). But I think this is almost impossible to trace all codepaths and guarantee that by the time `PrefixFilter` is called, the path it is given was canonicalized.